### DISCUSSION TITLE
Tiff to multiple jpegs using -write

### DISCUSSION THREAD

**Discussion by martimpassos** [ImageMagick/ImageMagick#2590] [Help] [answered]
When using 
```
magick file -set filename:fn "%t" -quality 100 -format jpg -write ./jpeg-hd/%[filename:fn].jpg +write -resize 1000x1000 ./jpeg-sd/%[filename:fn].jpg 
```
does the second jpg get saved from the input data or from the compressed -write output? What is the best practice to output multiple compressed files from a single uncompressed source?

**Comment by snibgo** [comment]
Your command includes:
```
... +write -resize 1000x1000  ...
```
... which is wrong. "+write" should be immediately followed by a filename, or a format such as "info:" or "txt:".

The documentation explains the difference between "-write" and "+write", although recently they both seem to behave the same.

**Reply by martimpassos** [reply]
I see. I thought +write reverted the data back after -write. So I'm looking for 
```
magick file -set filename:fn "%t" -quality 100 -format jpg +write ./jpeg-hd/%[filename:fn].jpg -resize 1000x1000 ./jpeg-sd/%[filename:fn].jpg 
```
and since +write doesn't restores the original data it could even be 
```
magick file -set filename:fn "%t" -quality 100 -format jpg -resize 1000x1000 +write ./jpeg-sd/%[filename:fn].jpg ./jpeg-hd/%[filename:fn].jpg 
```
?

**Reply by martimpassos** [reply]
I tried the latter and both output files are resized, so I guess they really are behaving the same

**Comment by snibgo** [comment, marked as answer]
According to the documentation, "-write" may change the image in memory. But "+write" makes a clone of the image, and may change that, but then removes the clone. Hence, "+write" won't change the image in memory. (This doesn't mean that "+write" will restore the image back to what it was before any other operations like "-resize".)

Recently, it seems that either "-write" doesn't change the image in memory, or it does a clone like "+write". But I haven't investigated this. To be safe, we should assume the documentation is correct, and that "-write" may change the image in memory.

This has no impact on your "-resize". That will change the image in memory.

**Reply by martimpassos** [reply]
Thank you for clarifying this. I'm not sure if there's a more thorough documentation somewhere, or if it's just me who got confused with the "To restore the image to its original state after writing it, use the +write filename option." statement [here](https://imagemagick.org/script/command-line-options.php#write)

**Comment by snibgo** [comment]
Yes, that documentation is misleading. It says "if there are additional options" but does that mean additional options before the write or after the write or what? I don't know.

> To restore the image to its original state after writing it, use the +write filename option.

"Original state" means "the state immediately before the write", not any previous state.
